### PR TITLE
chore(docs): add custom seed paths documentation

### PR DESCRIPTION
**Pull request by avallete**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

This PR update the documentation about how to split seeds, and document new `config.yml` options.

## What is the current behavior?

Currently, the documentation assumes a single seed.sql file in the supabase folder for seeding the database. The seeding process is not configurable.

## What is the new behavior?

- Removed references to creating an empty seed.sql file during project initialization.
- Updated the documentation to reflect the new feature that allows users to configure multiple seed files using glob patterns in config.toml.
- Added explanations on how the system processes seed files, including file sorting, deduplication, and logging of unmatched patterns.
- Enhanced descriptions of the db.seed.enabled and db.seed.sql_paths configuration options.

## Additional context

Following: https://github.com/supabase/cli/pull/2702

### CODE REVIEW

**Review by charislam — APPROVED** [review]

**Comment by sweatybridge** [comment]
Note: merge this only after next cli prod deploy, tentatively scheduled on 7 Oct.

File: apps/docs/content/guides/cli/local-development.mdx [hunk 1]
```diff
@@ -296,7 +296,7 @@ alter table
 
 Now that you are managing your database with migrations scripts, it would be great have some seed data to use every time you reset the database.
 
-For this, you can use the seed script in `supabase/seed.sql`. This file was automatically created when you ran [`supabase init`](/docs/reference/cli/usage#supabase-init)) at the beginning.
+For this by default, you can use the seed script in `supabase/seed.sql`.
```
Comment (sweatybridge) [inline apps/docs/content/guides/cli/local-development.mdx:299]:
```suggestion
For this, you can create a seed script in `supabase/seed.sql`.
```

File: apps/docs/content/guides/cli/seeding-your-database.mdx [hunk 2]
```diff
@@ -26,27 +26,35 @@ values
...
 
-The seed file can grow quite large, so you can split it up into multiple files. For example, you can create a `seed` folder and add a `countries.sql` file. Then you can add the following to your `seed.sql` file.
+For better modularity and maintainability, you can split your seed data into multiple files. For example, you can organize your seeds by table and include files such as `countries.sql` and `cities.sql`. Configure them in `config.toml` like so:
 
-```sql supabase/seed.sql
-\i seed/countries.sql
+```toml supabase/config.toml
+[db.seed]
+  enabled = true
+  sql_paths = ['./countries.sql', './cities.sql']
```
Comment (sweatybridge) [inline apps/docs/content/guides/cli/seeding-your-database.mdx:38]:
```suggestion
enabled = true
sql_paths = ['./countries.sql', './cities.sql']
```

File: apps/docs/content/guides/cli/seeding-your-database.mdx [hunk 3]
```diff
@@ -26,27 +26,35 @@ values
...
 
-co
-You can use the `COPY` command to import CSV data into your database. For example, you can create a `seed` folder and add a `countries.csv` file. Then you can add the following to your `seed.sql` file.
-
-```sql supabase/seed.sql
-COPY countries (name, code) FROM 'seed/countries.csv' WITH (FORMAT csv);
+```toml supabase/config.toml
+[db.seed]
+  enabled = true
+  sql_paths = ['./seeds/*.sql']
```
Comment (sweatybridge) [inline apps/docs/content/guides/cli/seeding-your-database.mdx:46]:
```suggestion
enabled = true
sql_paths = ['./seeds/*.sql']
```

File: apps/docs/content/guides/cli/seeding-your-database.mdx [hunk 4]
```diff
@@ -26,27 +26,35 @@ values
...
 ```
 
-\*/}
+<Admonition type="tip">
+
+The system processes the seed files in the order they are listed in the sql_paths array. If a glob pattern is used and matches multiple files, those files are sorted in lexicographic order to ensure consistent execution. Additionally:
```
Comment (sweatybridge) [inline apps/docs/content/guides/cli/seeding-your-database.mdx:51]:
```suggestion
The CLI processes seed files in the order they are declared in the `sql_paths` array. If a glob pattern is used and matches multiple files, those files are sorted in lexicographic order to ensure consistent execution. Additionally:
```

**Review by sweatybridge — APPROVED** [review]
just a few nits

**Pull request merged**